#!/usr/bin/env python3
PACKAGE = "controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('linearVel', double_t, 0, "linearVel", 1.0, 0.0, 1.4)
gen.add('angularVel', double_t, 0, "angularVel", 1.2, 0.0, 1.4)
gen.add('umbral', double_t, 0, "umbral", 0.03, 0.0, 1)

gen.add('kpX', double_t, 0, "kpX", 1.2, 0.0, 3.0)
gen.add('kpY', double_t, 0, "kpY", 1.2, 0.0, 3.0)
gen.add('kdX', double_t, 0, "kdX", 0.001, 0.0, 1.0)
gen.add('kdY', double_t, 0, "kdY", 0.001, 0.0, 1.0)

gen.add('xDes', double_t, 0, "xDes", 0.0,  -10.0, 10.0)
gen.add('yDes', double_t, 0, "yDes", 0.0,  -10.0, 10.0)

gen.add('h', double_t, 0, "h", 0.2,  0.01, 2.0)
gen.add('amplitud', double_t, 0, "amplitud", 0.8,  0.0, 10.0)
gen.add('longitud', double_t, 0, "longitud", 0.3,  0.0, 2.0)

exit(gen.generate(PACKAGE, "controller", "cf_dina2"))
